kibana-reports/public/components/main/report_details/report_details.tsx

```diff
@@ -121,7 +121,7 @@ export function ReportDetails(props) {
   const parseTimePeriod = (queryUrl: string) => {
     let timeString = queryUrl.substring(
       queryUrl.lastIndexOf('time:'),
-      queryUrl.lastIndexOf('))')
+      queryUrl.indexOf('))')
```

Comment:
This change doesn't seem to be specific for visualization and saved search, but why it was fine with dashboard before this fix?

Reply:
The url construction for visualization vs saved search vs dashboard is not the same

Reply:
I get that it's different but the time range part of the url should be the same, the only difference is dashboard#/view vs discover#/edit, as far as I can remember.

Reply:
This solution of locating a substring with `))` seems kind of hack for me. I'd rather use a more generic solution. Could you try this regex expression to match the time field `/time:\(from:.+,to:.+?\)/`